include/exadg/compressible_navier_stokes/postprocessor/hdf5_generator.h

```diff
@@ -0,0 +1,70 @@
+/*  ______________________________________________________________________
+ *
+ *  ExaDG - High-Order Discontinuous Galerkin for the Exa-Scale
+ *
+ *  Copyright (C) 2021 by the ExaDG authors
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *  ______________________________________________________________________
+ */
+
+#ifndef INCLUDE_EXADG_COMPRESSIBLE_NAVIER_STOKES_POSTPROCESSOR_HDF5_GENERATOR_H_
+#define INCLUDE_EXADG_COMPRESSIBLE_NAVIER_STOKES_POSTPROCESSOR_HDF5_GENERATOR_H_
+
+#include <exadg/postprocessor/hdf5_generator_base.h>
+
+namespace ExaDG
+{
+namespace CompNS
+{
+template<int dim>
+struct HDF5Data : public HDF5DataBase<dim>
+{
+  HDF5Data();
+
+  void
+  print(dealii::ConditionalOStream & pcout) const;
+
+  bool write_rho;
+  bool write_rho_u;
+  bool write_rho_E;
+};
+
+template<int dim, typename Number>
+class HDF5Generator : public HDF5GeneratorBase<dim, Number>
+{
+public:
+  using VectorType = typename HDF5GeneratorBase<dim, Number>::VectorType;
+
+  HDF5Generator(MPI_Comm const & comm);
+
+  void
+  setup(dealii::DoFHandler<dim> const & dof_handler_in,
+        dealii::Mapping<dim> const &    mapping_in,
+        HDF5Data<dim> const &           hdf5_data_in);
+
+private:
+  void
+  do_evaluate(VectorType const & solution) final;
+
+  HDF5Data<dim>                                       hdf5_data;
+  dealii::SmartPointer<dealii::DoFHandler<dim> const> dof_handler;
+  dealii::Vector<Number>                              hyperslab_data;
```

Comment:
Related to the above comment (https://github.com/exadg/exadg/pull/255/files#r964020314), I believe this variable should not be a member variable of this class.

Reply:
resolved with [7647d7b](https://github.com/exadg/exadg/pull/255/commits/7647d7b3beaddfbed1cba903da224c0bbb5f09a5)